skins/laika/tests/unit/components/shared/BankData.spec.ts

```diff
@@ -215,6 +216,52 @@ describe( 'BankData', () => {
 		expect( iban.text() ).toMatch( 'Test Bank' );
 	} );
 
+	it( 'renders bank code / BIC field when legacy bank account number is entered', async () => {
+		const wrapper = mount( BankData, {
+			localVue,
+			store: createStore(),
+			mocks: {
+				$t: () => {},
+			},
+		} );
+
+		let bankInfoInput = wrapper.find( 'input#bic' );
+		expect( bankInfoInput.isVisible() ).toBe( false );
+
+		wrapper.setData( { accountId: '123' } );
+		await wrapper.vm.$nextTick();
+
+		bankInfoInput = wrapper.find( 'input#bic' );
+		expect( bankInfoInput.isVisible() ).toBe( true );
+	} );
+
+	it( 'renders bank code / BIC field when valid IBAN was entered and validated', async () => {
```

Comment:
The way this test is written, it almost looks like an integration/browser test, simulating user input, server response and then desired state. I'm not sure we should go this route, because it will make the test brittle and we have a lot to set up. This test and other test can probably be simplified by setting up different store states and checking what's displayed. And by interacting with the form and check that the component dispatches the right actions to the store.

See https://vue-test-utils.vuejs.org/guides/using-with-vuex.html#mocking-with-modules (and the previous sections) for ideas on how to create a local store dummy for testing instead of using the `createStore` function.